lls/main.c

```diff
@@ -911,23 +920,18 @@ lls_stage2(void *arg)
 		}
 
 		/* Receive ICMP packets using IPv4 ACL filters. */
-		ret = register_icmp_acl_rules(&net_conf->back);
-		if (ret < 0)
-			return ret;
-	}
-
-	/* Receive ND packets using IPv6 ACL filters. */
-
-	if (lls_conf->nd_cache.iface_enabled(net_conf, &net_conf->front)) {
-		ret = register_icmp6_acl_rules(&net_conf->front);
```

Comment:
You moved the code above, but above it tests in the ARP cache, and here it tests in the ND cache.